### PR TITLE
Fhmobsdk 20.travis.build

### PR DESCRIPTION
**Pull request by corinnekrych**
This PR is for to support travis build
- add travis.yml
- share FH schema
- make unit tests passed in travis environment (vendorId is null in such env)

@cvasilak @wei-lee mind to review?

### CODE REVIEW

**Comment by wei-lee** [comment]
@corinnekrych  this looks good. Can we use travis to publish to Coacoapod automatically?

**Comment by corinnekrych** [comment]
@wei-lee @cvasilak you can do lot of things with travis, like running post/pre script etc... But I wonder do we really want to automatize the release to cocoapods?
Usually we do the push to cocoapods just before releasing a new version not for each PR.
what is your intend?

**Comment by wei-lee** [comment]
@corinnekrych  You are right, we are not intending to do a new release for every PR. I am wondering if we can do a release when a new tag is pushed, like what we did for the [android sdk](https://github.com/feedhenry/fh-android-sdk/blob/master/.travis.yml). This way we don't have to run any script manually, all we have to do is to push a tag. 
WDYT?

**Comment by corinnekrych** [comment]
Ah! that is clever let me dig in more closely
@wei-lee

**Comment by corinnekrych** [comment]
@wei-lee After further investigation it's not possible to automate the release process on cocoapods. since version 0.33, releasing pod goes through Trunk (https://guides.cocoapods.org/making/getting-setup-with-trunk.html). As a prerequisites, before releasing you need to register. The registration process is associated to a computer ("You do not have a password, only a per-computer session token.") and required an async email validation. 

The release process as described in https://cocoapods.org/?q=FH
could automate b and c part (gihub release and doc generation) with gh token (for auth) and some more scripting. we can track that part with a different ticket to discuss further.

**Comment by cvasilak** [comment]
> @wei-lee After further investigation it's not possible to automate the release process on cocoapods. since version 0.33, releasing pod goes through Trunk (https://guides.cocoapods.org/making/getting-setup-with-trunk.html). As a prerequisites, before releasing you need to register. The registration process is associated to a computer ("You do not have a password, only a per-computer session token.") and required an async email validation.

sounds reasonable 

> The release process as described in https://cocoapods.org/?q=FH
could automate b and c part (gihub release and doc generation) with gh token (for auth) and some more scripting. we can track that part with a different ticket to discuss further.

+1 let's track it separate and look further if we can get more automation on the release process

**Comment by cvasilak** [comment]
:eyes: on it

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: objective-c
+osx_image: xcode63
+xcode_workspace: fh-ios-sdk.xcworkspace
+xcode_scheme: FH
+xcode_sdk: iphonesimulator
+
+before_install:
+    - gem install cocoapods --no-rdoc --no-ri --no-document --quiet
```
Comment (cvasilak) [inline .travis.yml]: let's further add the notification step once we are about to merge [like here](https://github.com/aerogear/aerogear-ios-http/blob/master/.travis.yml#L10-L18)

**Comment by cvasilak** [comment]
tested on experimental travis build, and got green :green_heart: flag 
 https://travis-ci.org/cvasilak/fh-ios-sdk/builds/59375004

:+1: to merge.

**Comment by corinnekrych** [comment]
I've created:
- FHMOBSDK-65 to further automate process
- FHMOBSDK-66 to run the build on a matrix (iOS7, iOS8) like the android travis is doing.
Other PRs will come.
@wei-lee Good to merge?

**Comment by wei-lee** [comment]
@corinnekrych  :+1:

**Pull request merged**